src/components/Segmented/Segmented.tsx

```diff
@@ -0,0 +1,141 @@
+import { ReactElement, useCallback, useState } from 'react'
+import classnames from 'classnames'
+
+import { Option, OptionsAlignments } from './Segmented.types'
+import { getOptionKey, isDisabledOption, isString, isVerticalOption, resolveKey } from './Segmented.utils'
+import styles from './Segmented.module.css'
+
+const { Horizontal, Vertical } = OptionsAlignments
+const { segmented, segmentedOption, disabled, selected, vertical } = styles
+
+export type SegmentedProps = {
+
+  /**
+   * The option initially selected. Either one of the following:
+   *
+   * - string: the key of the initially selected option
+   * - number: the position in the array of the initially selected option
+   */
+  defaultValue?: string | number
+
+  /**
+   * Determines whether the segmented control is disabled.
+   */
+  isDisabled?: boolean
+
+  /**
+   * A callback function that is invoked when the selected option changes.
+   * @param {Option} option - The newly selected option.
+   */
+  onChange?: (option: Option) => void
+
+ /**
+   * An array of segmented options to be displayed. Either one of the following:
+   *
+   * - string
+   * - object: {
+   *
+   *     icon?: ReactNode, The icon of the option
+   *
+   *     label: ReactNode, The selectable text of the option
+   *
+   *     isDisabled?: boolean, Whether the option is clickable
+   *
+   *     key: string, The value associated with the option
+   *
+   *   }
```

Comment:
```suggestion
   * - `object: {`
   *
   *     `icon?: ReactNode` - The icon of the option
   *
   *     `label: ReactNode` - The selectable text of the option
   *
   *     `isDisabled?: boolean` - Whether the option is clickable
   *
   *     `key: string` - The value associated with the option
   *
   *   `}`
```

Reply:
If you don't like it go ahead and resolve the comment and write me so I can adjust it in my PR as well 🤙🏼

Reply:
Nice hint! What do you think about wrapping only types? sub fields are still emphasized correctly but we don't lose the classic format of Storybook.

Reply:
When you talk about types do you mean only `string`, `boolean`, and so on? That sounds okay to me, and would you also include the `object` that wraps everything?

Reply:
I followed your suggestion and now it is like this:

![image](https://github.com/mia-platform/design-system/assets/9254840/911a647a-ce04-452c-9e1e-716bdd9cd3c1)

I think it clear enough now